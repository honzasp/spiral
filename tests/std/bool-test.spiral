(program
  (import std)
  (import std.bool)
  (import std.test)

  (test "true" (lambda (t)
    (assert t true)))
  (test "false" (lambda (t)
    (assert-not t false)))
  (test "not" (lambda (t)
    (assert-not t (not true))
    (assert-not t (not 0))
    (assert t (not false))
    (assert-eqv t (not false) true)))
  (test "xor" (lambda (t)
    (assert t (xor true false))
    (assert t (xor false true))
    (assert-not t (xor true true))
    (assert-not t (xor false false)))))
