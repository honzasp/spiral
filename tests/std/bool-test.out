test true ok
test false ok
test not ok
test xor ok
